lib/services/platform-service.ts

```diff
@@ -217,6 +217,16 @@ export class PlatformService implements IPlatformService {
 			this.ensurePlatformInstalled(platform).wait();
 			let changesInfo = this.$projectChangesService.checkForChanges(platform);
 			if (changesInfo.hasChanges) {
+				// android build artifacts need to be cleaned up when switching from release to debug builds
+				if (platform.toLowerCase() === "android") {
```

Comment:
Thought about that earlier, but I don't think `cleanProject` should depend on conditions specific to the prepare phase, and the *android* test prevents unnecessary prepareInfo checks when preparing an ios project.